browser/src/control/jsdialog/Widget.TreeView.js

```diff
@@ -394,6 +405,19 @@ function _hasIcon(columns) {
 	return false;
 }
 
+function _createSelectionElement (parentContainer, treeViewData, entry, builder) {
+	var selectionElement;
+	switch (treeViewData.checkboxtype) {
+		case 'radio':
+			selectionElement = _createRadioButton(parentContainer, treeViewData, builder, entry);
+			break;
+		case 'checkbox':
+			selectionElement = _createCheckbox(parentContainer, treeViewData, builder, entry);
+			break;
+	}
```

Comment:
Core could send empty checkboxtype.
can we do this like: if (checkboxtype === radio) createRadio else createCheckbox ?

Reply:
i am afraid we can't do if else because some cases does not have neither checkbox nor radio.

Reply:
but this is done only when `entry.state` exists which means we need one of those

Reply:
@eszkadev yes you are right for those cases where no radio or checkbox the entry.state will be undefined and it will never call the _createSelectionElement

Reply:
Okay got it i am updating switch with if else